momepy/utils.py

```diff
@@ -470,20 +470,19 @@ def limit_range(vals, rng):
     array
         limited array
     """
+
     vals = np.asarray(vals)
-    if len(vals) > 2:
+
+    if len(vals) > 2 or not np.isnan(vals).all():
         if NumpyVersion(np.__version__) >= "1.22.0":
             method = dict(method="nearest")
         else:
             method = dict(interpolation="nearest")
         rng = sorted(rng)
-        if np.isnan(vals).any():
-            lower = np.nanpercentile(vals, rng[0], **method)
-            higher = np.nanpercentile(vals, rng[1], **method)
-        else:
-            lower = np.percentile(vals, rng[0], **method)
-            higher = np.percentile(vals, rng[1], **method)
```

Comment:
Then reason being this is performance. I know you can pass anything to `nanpercentile` but there's a performance hit doing that, compared to `percentile`. At there was when I wrote this.

Reply:
Ah, OK. I'll do more timing tests.

Reply:
Yep, using `nanpercentile` regardless of NaNs in the collection makes a performance difference. Good call.

```python
import momepy as mm
import numpy as np

%timeit mm.limit_range(np.array([0, 1, 2]), rng=(25, 75))
# 99.9 µs ± 5.32 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

%timeit mm.limit_range(np.array([np.nan, 1, 2]), rng=(25, 75))
# 115 µs ± 3.71 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

%timeit mm.limit_range(np.array([1, 2]), rng=(25, 75))
# 598 ns ± 12.9 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

%timeit mm.limit_range(np.array([np.nan, np.nan, np.nan]), rng=(25, 75))
# 2.69 µs ± 89.1 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```